sw/host/opentitanlib/src/test_utils/rpc.rs

```diff
@@ -9,46 +9,59 @@ use serde::Serialize;
 use std::io::Write;
 use std::time::Duration;
 
-use crate::io::uart::{Uart, UartError};
+use crate::io::console::{ConsoleDevice, ConsoleError};
 use crate::test_utils::status::Status;
 use crate::uart::console::{ExitStatus, UartConsole};
 
 // Bring in the auto-generated sources.
 include!(env!("ottf"));
 
-pub trait UartSend {
-    fn send(&self, uart: &dyn Uart) -> Result<()>;
-
-    fn send_with_crc(&self, uart: &dyn Uart) -> Result<()>;
+pub trait UartSend<T>
+where
+    T: ConsoleDevice + ?Sized,
+{
+    fn send(&self, device: &T) -> Result<()>;
+    fn send_with_crc(&self, device: &T) -> Result<()>;
 }
 
-impl<T: Serialize> UartSend for T {
-    fn send(&self, uart: &dyn Uart) -> Result<()> {
+impl<T, U> UartSend<T> for U
```

Comment:
maybe this should be `ConsoleSend` eventually? not sure how invasive changing this would be? if too much of a pain we could do it in a separate PR

Reply:
You're right. I initially planned to rename this in a separate PR, but after reviewing the codebase, the changes seemed less extensive than expected. I've now included a commit for the renaming in this PR.